astropy/timeseries/periodograms/lombscargle/implementations/main.py

```diff
@@ -218,6 +222,10 @@ def lombscargle(
         )
         kwds.update(f0=f0, df=df, Nf=Nf)
 
+    # if the grid is regular enable additional optimizations in the cython method
+    if method == "cython":
+        kwds.update(assume_regular_frequency=assume_regular_frequency)
```

Comment:
Wondering if this could additionally be checked using `_is_regular` even if not set explicitly, similar to how it's done in the 'fast' methods.

Reply:
It definitely could have been done, but I have no idea if it's worth it. Implicit check introducesna noticeable (but not too significant) overhead for nonregular grids, while that one generally tries to keep a good balance between uniform and nonuniform grids.

But again nonuniformly spaced ones have no practical purpose, so maybe the implicit check should've been added regardless. What do you think about it?

Btw branching in the Cython code is adding a noticeable overhead over the near-optimal implementation regardless, so it might actually be a good idea.